cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/UpdateTopicConfigurationRunnable.java

```diff
@@ -48,38 +57,107 @@
+    _allowCapacityEstimation = allowCapacityEstimation;
+    _concurrentInterBrokerPartitionMovements = SELF_HEALING_CONCURRENT_MOVEMENTS;
+    _concurrentLeaderMovements = SELF_HEALING_CONCURRENT_MOVEMENTS;
+    _executionProgressCheckIntervalMs = SELF_HEALING_EXECUTION_PROGRESS_CHECK_INTERVAL_MS;
+    _skipHardGoalCheck = SELF_HEALING_SKIP_HARD_GOAL_CHECK;
+    _replicaMovementStrategy = SELF_HEALING_REPLICA_MOVEMENT_STRATEGY;
+    _replicationThrottle = kafkaCruiseControl.config().getLong(ExecutorConfig.DEFAULT_REPLICATION_THROTTLE_CONFIG);
+    _excludeRecentlyDemotedBrokers = excludeRecentlyDemotedBrokers;
+    _excludeRecentlyRemovedBrokers = excludeRecentlyRemovedBrokers;
+    _dryRun = SELF_HEALING_DRYRUN;
+    _reason = reason;
+    _stopOngoingExecution = SELF_HEALING_STOP_ONGOING_EXECUTION;
+    _uuid = anomalyId;
+    _isTriggeredByUserRequest = false;
+  }
+
+
   @Override
-  protected OptimizationResult getResult() throws Exception {
-    if (_topicReplicationFactorChangeParameters != null) {
+  public OptimizationResult getResult() throws Exception {
+    if (_topicPatternByReplicationFactor != null) {
       return new OptimizationResult(
-          updateTopicReplicationFactor(_topicReplicationFactorChangeParameters.topicPatternByReplicationFactor(),
-                                       _topicReplicationFactorChangeParameters.goals(),
-                                       _topicReplicationFactorChangeParameters.skipRackAwarenessCheck(),
-                                       _topicReplicationFactorChangeParameters.modelCompletenessRequirements(),
-                                       _topicReplicationFactorChangeParameters.allowCapacityEstimation(),
-                                       _topicReplicationFactorChangeParameters.concurrentInterBrokerPartitionMovements(),
-                                       _topicReplicationFactorChangeParameters.concurrentLeaderMovements(),
-                                       _topicReplicationFactorChangeParameters.executionProgressCheckIntervalMs(),
-                                       _topicReplicationFactorChangeParameters.skipHardGoalCheck(),
-                                       _topicReplicationFactorChangeParameters.replicaMovementStrategy(),
-                                       _topicReplicationFactorChangeParameters.replicationThrottle(),
-                                       _topicReplicationFactorChangeParameters.excludeRecentlyDemotedBrokers(),
-                                       _topicReplicationFactorChangeParameters.excludeRecentlyRemovedBrokers(),
-                                       _topicReplicationFactorChangeParameters.dryRun(),
-                                       _topicReplicationFactorChangeParameters.reason(),
-                                       _topicReplicationFactorChangeParameters.stopOngoingExecution()),
+          updateTopicReplicationFactor(_topicPatternByReplicationFactor,
```

Comment:
If all parameters are local, can we avoid passing them to the method of the same class (e.g. see `RemoveBrokersRunnable#removeBrokers()`)?